python/client/cortex/util.py

```diff
@@ -26,15 +26,19 @@ def cli_config_dir() -> Path:
 
 
 @contextmanager
-def open_temporarily(path, mode):
-    Path(path).parent.mkdir(parents=True, exist_ok=True)
+def open_temporarily(path, mode, delete_parent: bool = False):
```

Comment:
What do you say about changing `delete_parent` to `delete_parent_if_empty` and then only removing the parent dir if it's empty only?